Makefile

```diff
@@ -1,59 +1,149 @@
+GOPKGS      ?= $(shell $(GOCMD) list $(MODVENDOR) ./... | grep -v /vendor)
+
+GOLDFLAGS   :="
+GOLDFLAGS   += -X $(PACKAGE)/internal/pkg/version.version=$(VERSION)
+GOLDFLAGS   += -X $(PACKAGE)/internal/pkg/version.commitHash=$(COMMIT_HASH)
+GOLDFLAGS   += -X $(PACKAGE)/internal/pkg/version.buildDate=$(BUILD_DATE)
+GOLDFLAGS   +="
+
+GOBUILD     ?= CGO_ENABLED=0 $(GOCMD) build $(MODVENDOR) -ldflags $(GOLDFLAGS)
+GORUN       ?= GOOS=$(GOOS) GOARCH=$(GOARCH) $(GOCMD) run $(MODVENDOR)
+
+# Binary versions
+GOLANGCI_VERSION  := v1.17.1
+GITCHGLOG_VERSION := 0.8.0
 
 .PHONY: all
 all: clean deps lint test build
 
-.PHONY: authors
-authors:
-	git log --all --format='%aN <%aE>' | sort -u | egrep -v noreply > AUTHORS
-
-.PHONY: build
-build: authors build-darwin-amd64 build-freebsd-amd64 build-linux-amd64 build-windows-amd64
+#########################
+## Development targets ##
+#########################
+.PHONY: clean
+clean: ## Clean workspace
+	@ $(MAKE) --no-print-directory log-$@
+	rm -rf ./$(BUILD_DIR)
 
-build-darwin-amd64:
-	GOOS=darwin GOARCH=amd64 $(GOBUILD) -o bin/darwin-amd64/$(NAME)
+.PHONY: vendor
+vendor: ## Install 'vendor' dependencies
+	@ $(MAKE) --no-print-directory log-$@
+	$(GOCMD) mod vendor
 
-build-freebsd-amd64:
-	GOOS=freebsd GOARCH=amd64 $(GOBUILD) -o bin/freebsd-amd64/$(NAME)
+.PHONY: verify
+verify: ## Verify 'vendor' dependencies
+	@ $(MAKE) --no-print-directory log-$@
+	$(GOCMD) mod verify
 
-build-linux-amd64:
-	GOOS=linux GOARCH=amd64 $(GOBUILD) -o bin/linux-amd64/$(NAME)
+.PHONY: lint
+lint: ## Run linter
+	@ $(MAKE) --no-print-directory log-$@
+	GO111MODULE=on golangci-lint run ./...
+
+.PHONY: fmt
+fmt: ## Format all go files
+	@ $(MAKE) --no-print-directory log-$@
+	goimports -w $(GOFILES)
+
+.PHONY: checkfmt
+checkfmt: RESULT = $(shell goimports -l $(GOFILES) | tee >(if [ "$$(wc -l)" = 0 ]; then echo "OK"; fi))
+checkfmt: SHELL := /usr/bin/env bash
+checkfmt: ## Check formatting of all go files
+	@ $(MAKE) --no-print-directory log-$@
+	@ echo "$(RESULT)"
+	@ if [ "$(RESULT)" != "OK" ]; then exit 1; fi
 
-build-windows-amd64:
-	GOOS=windows GOARCH=amd64 $(GOBUILD) -o bin/windows-amd64/$(NAME).exe
+.PHONY: test
+test: ## Run tests
+	@ $(MAKE) --no-print-directory log-$@
+	$(GOCMD) test $(MODVENDOR) -v $(GOPKGS)
 
-.PHONY: clean
-clean:
-	rm -rf ./bin
+###################
+## Build targets ##
+###################
+.PHONY: build
+build: clean ## Build binary for current OS/ARCH
+	@ $(MAKE) --no-print-directory log-$@
+	$(GOBUILD) -o ./$(BUILD_DIR)/$(GOOS)-$(GOARCH)/$(NAME)
+
+.PHONY: build-all
+build-all: GOOS      = linux darwin windows freebsd
+build-all: GOARCH    = amd64 arm
+build-all: clean ## Build binary for all OS/ARCH
+	@ $(MAKE) --no-print-directory log-$@
+	@ ./hack/build/build-all-osarch.sh "$(BUILD_DIR)" "$(NAME)" "$(VERSION)" "$(GOOS)" "$(GOARCH)" $(GOLDFLAGS)
+
+#####################
+## Release targets ##
+#####################
+.PHONY: release patch minor major
```

Comment:
Let‘s keep the pattern of having a .phony above each target.

Reply:
+1